mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/SystemCodeTest.java

```diff
@@ -266,5 +272,96 @@ public void testCodeValuesNotAssignedToTable() {
+
+       
+        final String dependentcodeName = Utils.randomNameGenerator("StateList", 5);
+        
+        
+ 
+        final Integer createResponseId = (Integer) CodeHelper.createCode(this.requestSpec, this.responseSpec, dependentcodeName, createCodeResponseParentId,
+                CodeHelper.RESPONSE_ID_ATTRIBUTE_NAME);
+        
+        // verify code created
+        final HashMap newCodeAttributes = (HashMap) CodeHelper.getCodeById(this.requestSpec, this.responseSpec, createResponseId, "");
+
+        Assert.assertNotNull(newCodeAttributes);
+        
+        final String codeValue3 = Utils.randomNameGenerator("Kerala", 5);
+        final String codeValue4 = Utils.randomNameGenerator("LV", 5);
+
+        final int codeValue3Position = 1;
+        final int codeValue4Position = 1;
+
+        final String codeDescription3 = "Description1";
+        final String codeDescription4 = "Description2";
+        
+        // create first code value
+        final Integer createCodeValueResponseId3 = (Integer) CodeHelper.createCodeValue(this.requestSpec, this.responseSpec,
+        		createResponseId, codeValue3, codeDescription3, codeValue3Position,  createCodeValueResponseId1, CodeHelper.SUBRESPONSE_ID_ATTRIBUTE_NAME);
+
+        // create second code value
+        final Integer createCodeValueResponseId4 = (Integer) CodeHelper.createCodeValue(this.requestSpec, this.responseSpec,
+        		createResponseId, codeValue4, codeDescription4, codeValue4Position, createCodeValueResponseId2, CodeHelper.SUBRESPONSE_ID_ATTRIBUTE_NAME);
+        
+        // verify two code values created
+
+        final List<HashMap> codeValuesList2 = (List) CodeHelper.getCodeValuesForCode(this.requestSpec, this.responseSpec,
+        		createResponseId, "");
+
+        assertEquals("Number of code values returned matches number created", 2, codeValuesList2.size());
+        
+        final HashMap codeValueChanges = (HashMap) CodeHelper.updateCodeValue(this.requestSpec, this.responseSpec, createResponseId,
+        		createCodeValueResponseId3, codeValue3 + "CHANGE", codeDescription3 + "CHANGE", 4, createCodeValueResponseId1, "changes");
+
+        assertEquals("Verify changed code value name", codeValueChanges.get("name"), codeValue3 + "CHANGE");
+
+        assertEquals("Verify changed code value description", codeValueChanges.get("description"), codeDescription3 + "CHANGE");
+        
+        // delete code value
+        Integer deletedCodeValueResponseId1 = (Integer) CodeHelper.deleteCodeValueById(this.requestSpec, this.generalResponseSpec,
+        		createResponseId, createCodeValueResponseId3, CodeHelper.SUBRESPONSE_ID_ATTRIBUTE_NAME);
+
+        // Verify code value deleted
+
+        final ArrayList<HashMap> deletedCodeValueAttributes1 = (ArrayList<HashMap>) CodeHelper.getCodeValueById(this.requestSpec,
+                this.generalResponseSpec, createResponseId, deletedCodeValueResponseId1, CommonConstants.RESPONSE_ERROR);
```

Comment:
use createCodeValueResponseId3 instead of deletedCodeValueResponseId1 for validation